src/averageAges.js

```diff
@@ -37,7 +50,14 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const filteredWomen = withChildren
+    ? people.filter(({ sex, name }) =>
+      sex === PERSON_SEX_FEMALE && people.some((child) =>
+        child.mother === name)
+    )
+    : people.filter(({ sex }) => sex === PERSON_SEX_FEMALE);
```

Comment:
```suggestion
  const filteredWomen = people.filter(({ sex, name }) => (
    sex === PERSON_SEX_FEMALE
    && (!withChildren || people.some((child) => child.mother === name))
  ));
```

Reply:
But even better would be to split this to simpler parts
<img width="1046" alt="image" src="https://github.com/mate-academy/js_average-ages/assets/86789161/fbb2e546-f9d9-4282-8527-2b7bac8adc9d">